### PR TITLE
feat: Add light/dark/auto mode theme setting at application level

### LINKED ISSUE #9030: Implementing light/dark/auto mode theme setting at application level
### Is your proposal related to a problem?
Currently we lacking an option to set light/dark/auto mode in application level

### Describe the solution you'd like
With this item we can introduce handling light/dark/auto mode in application level.

PRD- [Link](https://app.clickup.com/37484951/v/dc/13qycq-2704/13qycq-19336)
Figma design- [Link](https://www.figma.com/file/r3w4HYmTVfkm9BvL2IKuv7/Export-app?type=design&node-id=0%3A1&mode=design&t=rWgOP3mbwaQMHqpG-1)

### If the feature is approved, would you be willing to submit a PR?
Yes

### PR DESCRIPTION
**Pull request by nakulnagargade**
Closes #9030

### CODE REVIEW

File: frontend/src/Editor/Viewer/DesktopHeader.jsx [hunk 1]
```diff
@@ -19,6 +20,14 @@ const DesktopHeader = ({ showHeader, appName, changeDarkMode, darkMode, setAppDe
...
     shallow
   );
+  const { appMode } = useEditorStore(
+    (state) => ({
+      appMode: state.appMode,
+    }),
+    shallow
+  );
+
+  const showDarkModeToggle = useMemo(() => appMode === 'auto', [appMode]);
```
Comment (kavinvenkatachalam) [inline frontend/src/Editor/Viewer/DesktopHeader.jsx:30]: Just checking, can we move this logic to the above import? Like as follows,

`  const { appMode, showDarkModeToggle } = useEditorStore(
    (state) => ({
      appMode: state.appMode,
      showDarkModeToggle: appMode === 'auto'
    }),
    shallow
  );`

This way we can avoid `useMemo`. Please check the feasibility on this

File: frontend/src/Editor/Viewer/MobileHeader.jsx [hunk 2]
```diff
@@ -30,6 +31,13 @@ const MobileHeader = ({
     }),
     shallow
   );
+  const { appMode } = useEditorStore(
+    (state) => ({
+      appMode: state.appMode,
+    }),
+    shallow
+  );
+  const showDarkModeToggle = useMemo(() => appMode === 'auto', [appMode]);
```
Comment (kavinvenkatachalam) [inline frontend/src/Editor/Viewer/MobileHeader.jsx:40]: `const { appMode, showDarkModeToggle } = useEditorStore(
    (state) => ({
      appMode: state.appMode,
      showDarkModeToggle: state.appMode === 'auto'
    }),
    shallow
  );`

Check for the feasibility. This way we can avoid `useMemo`

File: frontend/src/Editor/Header/useAppDarkMode.js [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+    }),
+    shallow
+  );
+
+  const { setTheme } = useCurrentStateStore(
+    (state) => ({
+      setTheme: state.actions.setTheme,
+    }),
+    shallow
+  );
```
Comment (kavinvenkatachalam) [inline frontend/src/Editor/Header/useAppDarkMode.js:28]: Check once whether the current state updates will re-render this

File: frontend/src/Editor/Header/useAppDarkMode.js [hunk 4]
```diff
@@ -0,0 +1,54 @@
+import { useMemo } from 'react';
+import { shallow } from 'zustand/shallow';
+import { useEditorStore } from '../../_stores/editorStore';
+import { useAppDataStore } from '../../_stores/appDataStore';
```
Comment (kavinvenkatachalam) [inline frontend/src/Editor/Header/useAppDarkMode.js:4]:
```suggestion
import { useEditorStore } from '@/_stores/editorStore';
import { useAppDataStore } from '@/_stores/appDataStore';
```

File: frontend/src/Editor/Header/useAppDarkMode.js [hunk 5]
```diff
@@ -0,0 +1,54 @@
...
+  const isAppDarkMode = useMemo(() => {
+    let isDarkMode = false;
+    if (appMode === 'light') {
+      isDarkMode = false;
+    } else if (appMode === 'dark') {
+      isDarkMode = true;
+    } else {
+      isDarkMode = localStorage.getItem('darkMode') === 'true';
+    }
+    return isDarkMode;
```
Comment (kavinvenkatachalam) [inline frontend/src/Editor/Header/useAppDarkMode.js:44]:
```suggestion
    if (appMode === 'light') {
      return false;
    } else if (appMode === 'dark') {
      return true;
    } else {
      return localStorage.getItem('darkMode') === 'true';
    }
```

File: frontend/src/Editor/Header/useAppDarkMode.js [hunk 6]
```diff
@@ -0,0 +1,54 @@
+import { useMemo } from 'react';
```
Comment (kavinvenkatachalam) [inline frontend/src/Editor/Header/useAppDarkMode.js:1]: Can we move this to a hooks folder?

**Review by kavinvenkatachalam — CHANGES_REQUESTED** [review]
@nakulnagargade Requested few changes. Please take a look

File: frontend/src/Editor/Header/GlobalSettings.jsx [hunk 7]
```diff
@@ -380,7 +389,27 @@ export const GlobalSettings = ({
...
+                    defaultValue={appMode}
+                  >
+                    {APP_MODES.map((appMode) => (
+                      <ToggleGroupItem key={appMode.value} value={appMode.value}>
+                        {appMode.label}
+                      </ToggleGroupItem>
+                    ))}
+                  </ToggleGroup>
+                </div>
+              </div>
```
[the remark this answers is not in the document]
Comment (nakulnagargade) [reply frontend/src/Editor/Header/GlobalSettings.jsx]: Done.

File: frontend/src/Editor/Header/useAppDarkMode.js (hunk 3 above)
Comment (nakulnagargade) [reply frontend/src/Editor/Header/useAppDarkMode.js]: Checked, this is only rendering when there is a render on editor.

File: frontend/src/Editor/Container.jsx [hunk 8]
```diff
@@ -74,6 +74,7 @@ export const Container = ({
     }),
     shallow
   );
+  const { isAppDarkMode: darkMode } = useAppDarkMode();
```
[the remark this answers is not in the document]
Comment (nakulnagargade) [reply frontend/src/Editor/Container.jsx]: yes, moved this to Editor.

**Comment by nakulnagargade** [comment]
@kavinvenkatachalam All code comments are resolved.

**Review by kavinvenkatachalam — CHANGES_REQUESTED** [review]
@nakulnagargade Changing the global theme is not changing the exposed values. Please check it out

**Review by kavinvenkatachalam — CHANGES_REQUESTED** [review]
@johnsoncherian @nakulnagargade Still the exposed value is not correct. I made appMode as dark and when I try to change the editor mode, the exposed value is changing as well

image

File: frontend/src/Editor/Header/AppModeToggle.jsx [hunk 9]
```diff
@@ -0,0 +1,56 @@
+import React, { useContext } from 'react';
+import ToggleGroup from '@/ToolJetUI/SwitchGroup/ToggleGroup';
+import ToggleGroupItem from '@/ToolJetUI/SwitchGroup/ToggleGroupItem';
+import { useTranslation } from 'react-i18next';
+import useAppDarkMode from '@/_hooks/useAppDarkMode';
+import { useSuperStore } from '../../_stores/superStore';
+import { ModuleContext } from '../../_contexts/ModuleContext';
```
Comment (kavinvenkatachalam) [inline frontend/src/Editor/Header/AppModeToggle.jsx:7]:
```suggestion
import { useSuperStore } from '@/_stores/superStore';
import { ModuleContext } from '@/_contexts/ModuleContext';
```

File: frontend/src/Editor/Viewer/DesktopHeader.jsx [hunk 10]
```diff
@@ -1,4 +1,4 @@
-import React from 'react';
+import React, { useMemo } from 'react';
```
Comment (kavinvenkatachalam) [inline frontend/src/Editor/Viewer/DesktopHeader.jsx:1]:
```suggestion
import React from 'react';
```

File: frontend/src/_hooks/useAppDarkMode.js [hunk 11]
```diff
@@ -0,0 +1,43 @@
...
+  );
+
+  const { isTJDarkMode } = useAppDataStore(
+    (state) => ({
+      isTJDarkMode: state.isTJDarkMode,
+    }),
+    shallow
+  );
+
+  const handleAppModeChange = (appMode = 'auto') => {
```
Comment (kavinvenkatachalam) [inline frontend/src/_hooks/useAppDarkMode.js:22]: Can we warp it inside `useCallback`?

**Review by kavinvenkatachalam — APPROVED** [review]
LGTM

**Review by johnsoncherian — APPROVED** [review]

**Pull request merged**